src/Components/v2/ArtworkFilter/index.tsx

```diff
@@ -0,0 +1,306 @@
+  query ArtworkFilterQuery(
+    $acquireable: Boolean
+    $artist_id: String
+    $at_auction: Boolean
+    $attribution_class: [String]
+    $color: String
+    $for_sale: Boolean
+    $height: String
+    $inquireable_only: Boolean
+    $major_periods: [String]
+    $medium: String
+    $offerable: Boolean
+    $page: Int
+    $partner_id: ID
+    $price_range: String
+    $sort: String
+    $term: String!
+    $width: String
+  ) {
+    viewer {
+      ...ArtworkFilter_viewer
+        @arguments(
+          acquireable: $acquireable
+          artist_id: $artist_id
+          at_auction: $at_auction
+          attribution_class: $attribution_class
+          color: $color
+          for_sale: $for_sale
+          height: $height
+          inquireable_only: $inquireable_only
+          keyword: $term
+          major_periods: $major_periods
+          medium: $medium
+          offerable: $offerable
+          page: $page
+          partner_id: $partner_id
+          price_range: $price_range
+          sort: $sort
+          width: $width
+        )
+    }
+  }
+`
+
+interface ArtworkFilterProps {
+  viewer: ArtworkFilter_viewer
+  relay: RelayRefetchProp
+  term: string
+}
+
+const ArtworkFilter: React.FC<ArtworkFilterProps> = props => {
+  const { viewer, relay, term } = props
+  const tracking = useTracking()
+  const [isFetching, toggleFetching] = useState(false)
+  const [showMobileActionSheet, toggleMobileActionSheet] = useState(false)
+  const filterContext = useFilterContext()
+  const previousFilters = usePrevious(filterContext.filters)
+
+  /**
+   * Check to see if the mobile action sheet is present and prevent scrolling
+   */
+  useEffect(() => {
+    const setScrollable = doScroll => {
+      document.body.style.overflowY = doScroll ? "hidden" : "visible"
+    }
+    if (showMobileActionSheet) {
+      setScrollable(false)
+    }
+    return () => {
+      setScrollable(true)
+    }
+  }, [showMobileActionSheet])
+
+  /**
+   * Check to see if the current filter is different from the previous filter
+   * and trigger a reload.
+   */
+  useEffect(() => {
+    Object.entries(filterContext.filters).forEach(
+      ([filterKey, currentFilter]) => {
+        if (filterKey === "page") {
+          return
+        }
+        const previousFilter = previousFilters[filterKey]
+        const filtersHaveUpdated = !isEqual(currentFilter, previousFilter)
+
+        if (filtersHaveUpdated) {
+          fetchResults(filterKey)
+        }
+      }
+    )
+  }, [filterContext.filters])
+
+  function fetchResults(filterKey) {
+    if (isFetching) {
+      return
+    }
+
+    tracking.trackEvent({
+      action_type: AnalyticsSchema.ActionType.CommercialFilterParamsChanged,
+      current: filterContext.filters,
+      changed: {
+        [filterKey]: filterContext.filters[filterKey],
+      },
+    })
+
+    toggleFetching(true)
+
+    // FIXME: Re-enable fetch updates. Throws strange low-lever error ATM
+    return
```

Comment:
@pepopowitz - Hit a bit of a blocker here when making the request to update the grid based on filter change; an error comes back about `aggregations`. I don't have any more time to look at this ATM but feel free to poke at it a bit if you want. I can give it a look when I get back.